iOS/Flows/Conversation/Conversation Header/MembersViewController.swift

```diff
@@ -100,24 +109,48 @@ class MembersViewController: DiffableCollectionViewController<MembersCollectionV
         self.dataSource.appendItems([.member(member)], toSection: .members)
     }
 
+    /// A task for scrolling to a specific chat user.
+    private var scrollToUserTask: Task<Void, Never>?
+
     func scroll(to user: ChatUser) {
-        guard let controller = self.conversationController else { return }
+        self.scrollToUserTask?.cancel()
 
-        let member = Member(displayable: AnyHashableDisplayable(user),
-                            conversationController: controller)
-        guard let ip = self.dataSource.indexPath(for: .member(member)) else { return }
+        self.scrollToUserTask = Task { [weak self] in
+            // Wait for the data to finish loading before we try to scroll to a specific user.
+            await self?.loadDataTask?.value
 
-        self.collectionView.scrollToItem(at: ip, at: .centeredHorizontally, animated: true)
+            guard !Task.isCancelled,
+                let controller = self?.conversationController else { return }
+
+            let member = Member(displayable: AnyHashableDisplayable(user),
+                                conversationController: controller)
+            guard let ip = self?.dataSource.indexPath(for: .member(member)) else { return }
+
+            self?.collectionView.scrollToItem(at: ip, at: .centeredHorizontally, animated: true)
+        }
     }
 
     // MARK: - Data Loading
 
     override func getAnimationCycle(with snapshot: NSDiffableDataSourceSnapshot<MembersSectionType, MembersItemType>)
     -> AnimationCycle? {
+
+        // Center on the user who sent the most recent message.
+        var centeredIndexPath: IndexPath?
+        if let conversationController = conversationController,
+           let nonCurrentUserMessage = conversationController.messages.first(where: { message in
+               return !message.isFromCurrentUser
+           }) {
+            let user = nonCurrentUserMessage.author
+            let member = Member(displayable: AnyHashableDisplayable(user),
+                                conversationController: conversationController)
+            centeredIndexPath = snapshot.indexPathOfItem(.member(member))
+        }
```

Comment:
This isn't always correct, but it will generally land on the correct user. Even if this is wrong, it will be corrected later when scroll(to user) is called.